libs/teams/teams-chat-workflow-spring-boot-starter/src/main/java/org/finos/springbot/teams/messages/FileActivityHandler.java

```diff
@@ -0,0 +1,148 @@
+package org.finos.springbot.teams.messages;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.finos.springbot.teams.content.serialization.TeamsHTMLParser;
+import org.finos.springbot.teams.conversations.TeamsConversations;
+import org.finos.springbot.teams.state.TeamsStateStorage;
+import org.finos.springbot.workflow.actions.consumers.ActionConsumer;
+import org.finos.springbot.workflow.form.FormConverter;
+import org.finos.springbot.workflow.form.FormValidationProcessor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.microsoft.bot.builder.MessageFactory;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.Attachment;
+import com.microsoft.bot.schema.ResultPair;
+import com.microsoft.bot.schema.TextFormatTypes;
+import com.microsoft.bot.schema.teams.FileConsentCardResponse;
+import com.microsoft.bot.schema.teams.FileInfoCard;
+
+public class FileActivityHandler extends MessageActivityHandler {
```

Comment:
Please check the code coverage as a result of adding this.  I have a feeling there will need to be some unit tests added to make sure this works.

Reply:
Sure, I will check the code coverage and also add the unit testcases.

Reply:
committed testcases..

Reply:
Hi @robmoffat ,

Now code coverage is 72.53%